### DISCUSSION TITLE
Outdated genres-tree.yaml

### DISCUSSION THREAD

**Discussion by Emporea** [beetbox/beets#4765] [Q&A]
I wanted to make PR to the file [genres-tree.yaml](https://github.com/beetbox/beets/edit/master/beetsplug/lastgenre/genres-tree.yaml) and wanted to add some genres to it like Psychedelic trance, psytrance, psy or psybient, and more, since its missing some genres that are relevant to me.

- Then I asked myself how one genre can have multiple names in beets.
- Then I asked myself who even defines genres.
- Then I remembered: "ah yes, LastFM has popularity. so its just defined by users. nice"
- Then I looked at the LastFM page and found genres, but nothing about popularity. So what does the lastgenre option "min_weight: 10" actually say? Whats the scale? Where can I compare this number to on lastfm?
- So there are multiple names of the same genre, also on LastFM, which I just have to accept. Or is there an elegant way with beets to define my own "alias" genres? 
- I then looked at the genres-tree.yaml file again and wondered about it for some more time as I was starring at my pull request button. It seems outdated. Cant we just API request *ALL* genres from LastFM? The answer I found was a no.

Does someone have an idea on how we can update the genres-tree.yaml? I dont want to do this manually for my music... I just want to use whats already there: LastFM's genres by popularity (what ever this is, maybe its only visible with api requests and not on the website)

**Comment by JOJ0** [comment]
That is not easy to answer questions but I'll  add my thoughts since I recently started to tinker with the lastgenre plugin. Actually I stole some ideas and even code from https://github.com/yetanothernerd/whatlastgenre (which is a wonderful plugin on its own but has some things which dont suit my needs but you should definitely try it, if you havent yet!)


> - So there are multiple names of the same genre, also on LastFM, which I just have to accept. Or is there an elegant way with beets to define my own "alias" genres? 

Have a look at my branch https://github.com/beetbox/beets/compare/master...JOJ0:beets:all_in to see the alias and regex feature. I have several things improved and fixed with the lastgenre plugin, it's still a mess and not ready for a PR yet. Also some fixes required changes even down to beets library models. Too much to explain for now. Maybe start here

https://github.com/beetbox/beets/commit/03848c6bb948dbc3cbd9e9f82009c1df1366ed7f

https://github.com/beetbox/beets/commit/a924e584a049d3b4dec0b3641da9495210dbf7c8

and then read through the history up to the latest commit. Probably reading through my commit messages is enough to get were I'm heading with my ideas.

Or have a look at the whatlastgenre docs to see how aliases can be handled:

http://yetanothernerd.github.io/whatlastgenre/

https://github.com/YetAnotherNerd/whatlastgenre/blob/master/wlg/data/tags.txt


As said I like the approach and want to bring those features over to the lastgenre plugin.

> - I then looked at the genres-tree.yaml file again and wondered about it for some more time as I was starring at my pull request button. It seems outdated. Cant we just API request *ALL* genres from LastFM? The answer I found was a no.



IMHO genres _and_ there aliases are a rather personal thing and users opinions and ideas are differing a lot, thus no matter if static or pulled from last.fm, whitelist and genres tree files, it's always just a start for the user to work from there.
